exporter/sumologicexporter/config.go

```diff
@@ -0,0 +1,118 @@
+	URL string `mapstructure:"url"`
+	// Option to enable compression (default true)
+	Compress bool `mapstructure:"compress"`
+	// Compression encoding format, either gzip or deflate (default gzip)
+	CompressEncoding string `mapstructure:"compress_encoding"`
+	// Max HTTP request body size in bytes before compression (if applied).
+	// By default 1MB is recommended.
+	MaxRequestBodySize int `mapstructure:"max_request_body_size"`
+
+	// Logs related configuration
+	// Format to post logs into Sumo. (default json)
+	//   * text - Logs will appear in Sumo Logic in text format.
+	//   * json - Logs will appear in Sumo Logic in json format.
+	LogFormat string `mapstructure:"log_format"`
+
+	// Metrics related configuration
+	// The format of metrics you will be sending, either graphite or carbon2 or prometheus (Default is carbon2)
+	MetricFormat string `mapstructure:"metric_format"`
+
+	// List of regexes for attributes which should be send as metadata
+	MetadataFields []string `mapstructure:"metadata_attributes"`
+
+	// Sumo specific options
+	// Desired source category.
+	// Useful if you want to override the source category configured for the source.
+	SourceCategory string `mapstructure:"source_category"`
+	// Desired source name.
+	// Useful if you want to override the source name configured for the source.
+	SourceName string `mapstructure:"source_name"`
+	// Desired host name.
+	// Useful if you want to override the source host configured for the source.
+	SourceHost string `mapstructure:"source_host"`
+	// Name of the client
+	Client string `mapstructure:"client"`
+
+	exporterhelper.TimeoutSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	exporterhelper.QueueSettings   `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings   `mapstructure:"retry_on_failure"`
+}
+
+// CreateDefaultTimeoutSettings returns default timeout settings
+func CreateDefaultTimeoutSettings() exporterhelper.TimeoutSettings {
+	return exporterhelper.TimeoutSettings{
+		Timeout: defaultTimeout * time.Second,
+	}
+}
+
+const (
+	// TextFormat represents log_format: text
+	TextFormat string = "text"
+	// JSONFormat represents log_format: json
+	JSONFormat string = "json"
+	// GraphiteFormat represents metric_format: text
+	GraphiteFormat string = "graphite"
+	// Carbon2Format represents metric_format: json
+	Carbon2Format string = "carbon2"
+	// PrometheusFormat represents metric_format: json
+	PrometheusFormat string = "prometheus"
+	// GZIPCompression represents compress_encoding: gzip
+	GZIPCompression string = "gzip"
+	// DeflateCompression represents compress_encoding: deflate
+	DeflateCompression string = "deflate"
+	// MetricsPipeline represents metrics pipeline
+	MetricsPipeline string = "metrics"
+	// LogsPipeline represents metrics pipeline
+	LogsPipeline string = "logs"
+	// defaultTimeout
+	defaultTimeout time.Duration = 55
```

Comment:
This might be used incorrectly, who would have thought that this is a `defaultTimeout` of `55ns` 👍🏻 

```suggestion
	defaultTimeout time.Duration = 55 * time.Second
```